src/systems/systemItemsDialog.component.tsx

```diff
@@ -31,10 +63,47 @@ const SystemItemsDialog = React.memo((props: SystemItemsDialogProps) => {
   const [parentSystemId, setParentSystemId] = React.useState<string | null>(
     props.parentSystemId
   );
+  const [usageStatues, setUsageStatues] = React.useState<UsageStatuesType[]>(
+    []
+  );
+
+  const [usageStatuesErrors, setUsageStatuesErrors] = React.useState<
+    UsageStatuesErrorType[]
+  >([]);
+
+  const [aggregatedCellUsageStatus, setAggregatedCellUsageStatus] =
+    React.useState<Omit<UsageStatuesType, 'item_id'>[]>([]);
+
+  const [placeIntoSystemError, setPlaceIntoSystemError] = React.useState(false);
+  React.useEffect(() => {
+    if (open) {
+      const initialUsageStatues: UsageStatuesType[] = selectedItems.map(
+        (item) => ({
+          item_id: item.id,
+          catalogue_item_id: item.catalogue_item_id,
+          usageStatus: '',
+        })
+      );
+
+      const initialUsageStatuesErrors: UsageStatuesErrorType[] =
+        selectedItems.map((item) => ({
+          item_id: item.id,
+          catalogue_item_id: item.catalogue_item_id,
+          error: false,
+        }));
+      setUsageStatues(initialUsageStatues);
+      setUsageStatuesErrors(initialUsageStatuesErrors);
+    }
+  }, [open, selectedItems]);
+
   React.useEffect(() => {
     setParentSystemId(props.parentSystemId);
   }, [props.parentSystemId]);
 
+  React.useEffect(() => {
+    setPlaceIntoSystemError(false);
+  }, [parentSystemId]);
```

Comment:
They can't as the first one sets the current location in the dialog to the same as the system location and the one I create clears the error when the user navigates to a different system within the dialog.

Reply:
How about something like
```
const changeParentSystemId = (parentSystemId: string) => {
    setParentSystemId(parentSystemId)
    setPlaceIntoSystemError(false);
}
```
And replace all setParentSystemId's with this. It will stop the 2 consectutive re-renders every time the user navigates and avoids having anothe useEffect.